.github/workflows/cygwin-windows.yml

```diff
@@ -0,0 +1,79 @@
+# Based on https://github.com/mit-plv/fiat-crypto/blob/master/.github/workflows/coq-windows.yml
+name: CI (Cygwin/Windows)
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  build:
+
+    runs-on: windows-latest
+
+    env:
+      CFLAGS: "--coverage -O2 -g"
+      CXXFLAGS: "--coverage -O2 -g"
+      LDFLAGS: "--coverage"
+      # default config flags: enable debug asserts
+      CONFIGFLAGS: "--enable-debug"
+      COVERALLS_PARALLEL: true
+      TEST_SUITES: testinstall
+
+    steps:
+    # This sets git to use Linux line endings. While GAP should allow
+    # Windows line endings in GAP files, we currently can't build GAP if the
+    # whole source has Windows line endings
+    - name: Set git to use LF
+      run: |
+        git config --global core.autocrlf false
+        git config --global core.eol lf
+
+    - uses: actions/checkout@v2
+    - name: Download cygwin
+      run: |
+        & cmd /c 'nslookup www.cygwin.com 2>&1'
+        while ($true)
+        {
+            try
+            {
+                (New-Object Net.WebClient).DownloadFile('http://www.cygwin.com/setup-x86_64.exe', 'setup-x86_64.exe')
+                break
+            }
+            catch
+            {
+                Write-Host "There is an error during package downloading:`n $_"
+            }
+        }
+        & cmd /c 'nslookup www.cygwin.com 8.8.8.8 2>&1'
+      shell: powershell
+    - name: Install cygwin
+      run: |
+        @ECHO ON
+        SET CYGROOT=C:\cygwin64
+        SET CYGCACHE=%CYGROOT%\var\cache\setup
+        setup-x86_64.exe -qnNdO -R %CYGROOT% -l %CYGCACHE% -s %CYGMIRROR% -P wget,git,gcc,gcc-g++,gcc-core,m4,libgmp-devel,make,automake,libtool,autoconf,autoconf2.5,zlib-devel,libreadline-devel
+        SET TARGET_ARCH=x86_64-w64-mingw32
+        SET CD_MFMT=%cd:\=/%
```

Comment:
Out of curiosity, what could the FMT part stand for?

Reply:
that is an excellent question, googling, it looks like a bit I should have tried removing, let me try :)